library/hawkular_alerts_group_dampening.py

```diff
@@ -0,0 +1,297 @@
+            group_dampenings_dicts_list = [vars(dampening) for dampening in group_dampenings]
+        except Exception as e:
+            self.module.fail_json(msg="Failed to list group dampenings. Error: {error}".format(error=e))
+        return dict(
+            msg="Successfully listed group dampenings",
+            changed=self.changed,
+            group_dampenings=group_dampenings_dicts_list)
+
+    def update_required(self, group_id, desired_dampening):
+        """ Checks whether an update is required for the group trigger
+
+            Returns:
+               True, if an update is required for the dampening, False otherwise
+        """
+        current_dampenings = self.client.list_dampenings(group_id)
+        for current_dampening in current_dampenings:
+            if current_dampening.trigger_mode == desired_dampening["trigger_mode"]:
+                for key in desired_dampening.keys():
+                    if desired_dampening[key] != getattr(current_dampening, key):
+                        return True
+        return False
+
+    def update_group_dampening(self, group_id, dampening):
+        """ Updates a group trigger dampening in Hawkular Alerts
+
+            Returns:
+                whether or not a change took place and a short message
+                describing the operation executed
+        """
+        dampening_object = hawkular.alerts.Dampening(dampening)
+        try:
+            self.client.update_group_dampening(group_id, dampening["dampening_id"], dampening_object)
+        except Exception as e:
+            self.module.fail_json(msg="Failed to update dampening. Error: {error}".format(error=e))
+        self.changed = True
+
+    def create_group_dampening(self, group_id, dampening):
+        """ Creates a group trigger dampening in Hawkular Alerts
+
+            Returns:
+                whether or not a change took place and a short message
+                describing the operation executed
+        """
+        new_dampening = hawkular.alerts.Dampening(dampening)
+        try:
+            self.client.create_group_dampening(group_id, new_dampening)
+        except Exception as e:
+            self.module.fail_json(msg="Failed to create dampening. Error: {error}".format(error=e))
+        self.changed = True
+
+    def create_or_update_group_dampenings(self, group_id, dampenings):
+        """ Creates or updates group trigger dampening in Hawkular Alerts
+
+            Returns:
+                whether or not a change took place and a short message
+                describing the operation executed
+        """
+        if not self.group_trigger_exist(group_id):
+            self.module.fail(msg="Group trigger (group_id} does not exist ".format(group_id=group_id))
+        messages = []
+        for dampening in dampenings:
+            if self.dampening_exist(group_id, dampening.get("dampening_id")):
+                if self.update_required(group_id, dampening):
```

Comment:
> IIUC this does 2*N list_dampenings() requests. Can't you list once, then match them up?

I changed the logic, only querying once as you suggested

> n fact, AFAICT this module handles each dampening independently,
> so you could equivalently have made a module dealing with a single group dampening?

That's correct, but I'm allowing to pass 2 dampenings if the user prefers to do so